### DISCUSSION TITLE
A Not-Roadmap of Known/Open TODOs

### DISCUSSION THREAD

**Discussion by JWCS** [beartype/beartype#425] [General]
For the purpose of bringing awareness to the myriad of not gotten to TODOs, with no promise of their timeline, just their categorial existence.
# Docos
- [ ] #422
- [ ] #408
- [ ] #357
- [ ] #329
- [ ] #326
- [ ] #311
- [ ] #270
- [ ] #215
- [ ] #114
- [ ] #410
- [ ] #430
- [ ] #432
- [ ] #435
- [ ] #446 
- [ ] #450
- [ ] #477

## Docos: Broken 3rd Parties
- [ ] #436
- [ ] #465 

# Bugs
- [ ] #414
- [ ] #154
- [ ] #312
- [ ] #252
- [ ] #189
- [ ] #190
- [ ] #447 
- [ ] #478

# More Typing
- [ ] #53 (Just TypeDicts?)
- [ ] #423
- [ ] #373
- [ ] #391
- [ ] #350
- [ ] #128
- [ ] #364
- [ ] #297
- [ ] #319
- [ ] #314
- [ ] #219
- [ ] #295
- [ ] #236
- [ ] #238
- [ ] #223
- [ ] #119
- [ ] #141
- [ ] #54
- [ ] #445 

# Upcoming Python Typing Changes
- [ ] #440
- [ ] #459 

# QOL
- [ ] #397
- [ ] #393
- [ ] #369
- [ ] #336
- [ ] #184
- [ ] #426 (AKA eat pydantic)
- [ ] #429 
- [ ] #431
- [ ] #437 

# Strategies/Config
- [ ] #385
- [ ] #349
- [ ] #279
- [ ] #254
- [ ] #60

# Other Features
- [ ] #212
- [ ] #324
- [ ] #322
- [ ] #366
- [ ] #355
- [ ] #320
- [ ] #306
- [ ] #291
- [ ] #261
- [ ] #262
- [ ] #272
- [ ] #242
- [ ] #237 (typing backport)
- [ ] #192
- [ ] #196
- [ ] #191
- [ ] #193
- [ ] #187
- [ ] #131
- [ ] #116
- [ ] #424
- [ ] #430
- [ ] #438
- [ ] #439

# Changelog:
2024-08-29 Original List
2025-01-17 Removed 2; Added 19: 426, 429, 430, 431, 432, 435, 436, 437, 438, 439, 440, 445, 446, 447, 450, 459, 465, 477, 478; and for the record, 21 issues were opened and closed in the meantime as well.

**Comment by leycec** [comment]
**Woah.** It... it's *not* looking good, is it? Issues grow monotonically and hyperexponentially with time as users pound their fists on GitHub's window pane. Meanwhile, issue resolution shrinks monotonically and hyperexponentially with time as @leycec focuses on fun but ultimately unrelated stuff that nobody asked for like type hint inference.

![](https://media1.tenor.com/m/F02lwzSKPSAAAAAd/horse-man-drunk-carrying-a-horse.gif)
<sup>*bottom: @leycec. top: GitHub.*</sup>

**Reply by JWCS** [reply]
![image](https://github.com/user-attachments/assets/39767800-e1f9-42ee-89a9-db6d78a296d4)
Actually, it could be worse. The growth seems to only be _weakly_ non-linear, and only grow by ~4 (std dev 4) open issues per quarter. 
You can do it!                                                              <sub>(And it has to be you, because I tried for a month to PR, and was too lost).</sub>

**Reply by leycec** [reply]
> The growth seems to only be *weakly* non-linear...

I like what I'm hearing.

> ...and only grow by ~4 (std dev 4) open issues per quarter.

Tonight, we celebrate. At the current rate of only weakly non-linear growth, the @beartype issue tracker is now projected to succumb to its own corpulence sometime this century. Until then, the gravy train(wreck) continues.

> You can do it!

I probably can't. But that's okay! Total systemic failure is always an option, too. :smile: 

> ...because I tried for a month to PR, and was too lost...

**This saddens me muchly.** If you'd like to try again, <sup>*...you are now running away, aren't you? i can see you over there, you know!*</sup> I'd be *delighted* to gently detail exactly how to make all of your utopian PR dreams come true. With great details come both great understanding and great exhaustion – and one of those things is good.

Together, we can build a better @beartype. Together, we can derail this trainwreck before any other innocent codebases have to suffer. *Won't someone think of the bugs?*
